one-ui/src/main/java/com/marklogic/hub/oneui/services/EnvironmentService.java

```diff
@@ -15,21 +15,29 @@
  */
 package com.marklogic.hub.oneui.services;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.impl.Versions;
 import com.marklogic.hub.oneui.models.EnvironmentInfo;
+import com.marklogic.hub.oneui.models.HubConfigSession;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
+import java.io.*;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.prefs.Preferences;
 
 @Service
 public class EnvironmentService {
 
+    @Autowired
+    private HubConfigSession hubConfig;
```

Comment:
Couldn't you depend on HubProject instead, since that's not specific to a user?

Reply:
No, I tried autowiring HubProject object here but its fields (projDir etc) are not populated and all are null and had to get  the project from HubConfigSession